doc/contributing.rst

```diff
@@ -523,16 +427,40 @@ General commit rules
 
       #!/bin/sh
       git diff --check --cached || exit $?
-5. Describe your commit and use your common sense.
-   Example commit message::
+5. Describe your commit and use your common sense. Commit message should
+   contain enough information in order for end user to understand what is
+   going on by looking at the github log.
 
-     Fixes #123; refs #124
+   Example of bad commit message::
```

Comment:
My favorite edit so far, I can believe this has been "recommended" in the contribution guide

Reply:
Ooph, that's for smashing this.